core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -113,14 +125,21 @@ public void testReplaceTxnBuilder() throws Exception {
 
     table = catalog.loadTable(tableIdent);
     Assertions.assertThat(table.currentSnapshot()).isNull();
-    PartitionSpec v1Expected =
-        PartitionSpec.builderFor(table.schema())
-            .alwaysNull("data", "data_bucket")
-            .withSpecId(1)
-            .build();
-    Assertions.assertThat(table.spec())
-        .as("Table should have a spec with one void field")
-        .isEqualTo(v1Expected);
+
+    if (formatVersion == 1) {
+      PartitionSpec v1Expected =
+          PartitionSpec.builderFor(table.schema())
+              .alwaysNull("data", "data_bucket")
+              .withSpecId(1)
+              .build();
+      Assertions.assertThat(table.spec())
+          .as("Table should have a spec with one void field")
+          .isEqualTo(v1Expected);
+    } else {
+      Assertions.assertThat(table.spec().isUnpartitioned())
```

Comment:
This is actually a valid check on both versions now, unless we want to be sure to check for the void transform
https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/PartitionSpec.java#L89-L95

Reply:
I feel like it is better to highlight the difference in this case.